src/Tuura/Concept/FSM/Translation.hs

```diff
@@ -165,8 +188,8 @@ createArcs xs = zipWith3 createArc makeSrcEncs makeDestEncs activeTransitions
           invert = liftM2 Transition signal (not . newValue)
           activeTransitions = (map snd .  addMissingSignals) xs
 
-createInvariants :: Ord a => Invariant (Transition a) -> [a] -> [[Tristate]]
-createInvariants (NeverAll es) allSigns = expand (encode newTransitions)
+getInvariantStates :: Ord a => Invariant (Transition a) -> [a] -> [[Tristate]]
+getInvariantStates (NeverAll es) allSigns = expand (encode newTransitions)
     where newTransitions = transX ++ (concatMap (map (\s -> TransitionX { msignal = s, mnewValue = triX })) missingSignals)
           transX = map toTransitionX es
           missingSignals = map (allSigns \\) (onlySignals [es])
```

Comment:
Nothing specific -- just looks like this could be optimised. If you can't -- leave it as is.